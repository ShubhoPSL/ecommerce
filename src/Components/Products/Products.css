.product-main-div {
  @apply py-10 px-2 bg-yellow-300;
}

.product-heading {
  @apply text-lg font-serif font-medium;
}

.product-card-container {
  @apply grid grid-cols-1 xl:grid-cols-5 gap-0 px-2;
}

.error {
  @apply text-center text-4xl text-red-600 p-52;
}

.loading {
  @apply text-center text-5xl
  text-green-500 p-52;
}
