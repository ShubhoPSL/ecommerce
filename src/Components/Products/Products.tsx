import React, { useState, useEffect } from "react";
import { useRecoilState } from "recoil";
import Card from "../Card/Card";
import { allProducts } from "../States/AllProducts";

function Products() {
  const [products, setProducts] = useRecoilState(allProducts);
  const [error, setError] = useState(false);
  const [loading, setLoading] = useState(true);

  function fetchProducts() {
    fetch("https://fakestoreapi.com/products?limit=10")
      .then((res) => res.json())
      .then(
        (data) => {
          setLoading(false);
          setProducts(data);
        },
        (error) => {
          setLoading(false);
          setError(true);
        }
      );
  }
  useEffect(() => {
    fetchProducts();
  });

  if (error) return <div className="error">404 Not Found!</div>;
  if (loading)
    return (
      <div className="loading">Please Wait! Something is cooking up...</div>
    );

  return (
    <div className="product-main-div">
      <h2 className="product-heading">All Products</h2>
      <hr />
      <div className="product-card-container">
        {products &&
          products[0].id !== 0 &&
          products.map((product) => (
            <Card product={product} key={product.id} />
          ))}
      </div>
    </div>
  );
}

export default Products;
