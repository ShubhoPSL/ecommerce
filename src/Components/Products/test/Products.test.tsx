import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { RecoilRoot } from "recoil";
import Products from "../Products";

describe("Products component is being tested", () => {
  const NewProducts = () => {
    return (
      <RecoilRoot>
        <BrowserRouter>
          <Products />
        </BrowserRouter>
      </RecoilRoot>
    );
  };

  test("should have text All Products", () => {
    render(<NewProducts />);
    expect(screen.getByText("All Products")).toBeInTheDocument();
  });
  describe("Testing product rendering", () => {
    test("should have Product with id - 1", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 1")).toBeInTheDocument();
    });
    test("should have Product with id - 2", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 2")).toBeInTheDocument();
    });
    test("should have Product with id - 3", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 3")).toBeInTheDocument();
    });
    test("should have Product with id - 4", () => {
      render(<NewProducts />);
      expect(screen.getByText("Fav Item - 4")).toBeInTheDocument();
    });
    test("should have Product with id - 5", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 5")).toBeInTheDocument();
    });
    test("should have Product with id - 6", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 6")).toBeInTheDocument();
    });
    test("should have Product with id - 7", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 7")).toBeInTheDocument();
    });
    test("should have Product with id - 8", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 8")).toBeInTheDocument();
    });
    test("should have Product with id - 9", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 9")).toBeInTheDocument();
    });
    test("should have Product with id - 10", () => {
      render(<NewProducts />);
      expect(screen.getByText("Item - 10")).toBeInTheDocument();
    });
  });

  test("should match with snapshot of Products module", () => {
    const tree = renderer.create(<NewProducts />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
