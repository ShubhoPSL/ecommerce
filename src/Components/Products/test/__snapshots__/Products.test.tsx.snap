// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products component is being tested should match with snapshot of Products module 1`] = `
<div
  className="loading"
>
  Please Wait! Something is cooking up...
</div>
`;
